test/e2e/federated-service.go

```diff
@@ -260,17 +275,16 @@ func createService(fcs *federation_release_1_3.Clientset, clusterClientSets []*r
 		},
 	}
 	nservice, err := fcs.Core().Services(namespace).Create(service)
-	framework.Logf("Trying to create service %q in namespace %q", service.ObjectMeta.Name, service.ObjectMeta.Namespace)
 	Expect(err).NotTo(HaveOccurred())
 	for _, cs := range clusterClientSets {
 		waitForFederatedServiceShard(cs, namespace, nservice, 1)
 	}
 }
 
-func discoverService(f *framework.Framework, name string, exists bool) {
-	command := []string{"nslookup", name}
+func discoverService(f *framework.Framework, name string) {
+	command := []string{"sh", "-c", fmt.Sprintf("until nslookup '%s'; do sleep 1; done", name)}
```

Comment:
It's the write way as long as the next argument is a single string representing a bash command.  Before we had "sh" "-c" "nslookup" "foo", and when you do that, the "foo" goes nowhere.

There is an endless loop here if nslookup never succeeds.  But the Eventually has a timeout, and when we reach that timeout, the test fails and we'll delete the pod on the way out of the test.